subprojects/ide/src/main/groovy/org/gradle/plugins/ide/eclipse/EclipseWtpPlugin.groovy

```diff
@@ -67,53 +69,60 @@ class EclipseWtpPlugin extends IdePlugin {
+            }
+        }
+
         project.plugins.withType(JavaPlugin) {
-            def deleteWtpDependentModule = []
             eclipseModel.classpath.file.whenMerged { Classpath classpath ->
                 if (hasWarOrEarPlugin(project)) {
                     return
                 }
-
-                def minusFiles = eclipseModel.wtp.component.minusConfigurations*.files?.flatten() ?: project.files()
-                def libFiles = eclipseModel.wtp.component.libConfigurations*.files?.flatten() ?: project.files()
-                for (entry in classpath.entries) {
-                    if (!(entry instanceof AbstractLibrary)) {
-                        continue;
-                    }
-                    if (minusFiles.contains(entry.library.file) || !libFiles.contains(entry.library.file)) {
-                        // Mark this library as not required for deployment
-                        entry.entryAttributes[AbstractClasspathEntry.COMPONENT_NON_DEPENDENCY_ATTRIBUTE] = ''
-                    } else {
-                        // '../' and '/WEB-INF/lib' both seem to be correct (and equivalent) values here
-                        //this is necessary so that the depended upon projects will have their dependencies
-                        // deployed to WEB-INF/lib of the main project.
-                        entry.entryAttributes[AbstractClasspathEntry.COMPONENT_DEPENDENCY_ATTRIBUTE] = '../'
-                        deleteWtpDependentModule << entry.path
-                    }
-                }
+                adjustClasspath(classpath)
             }
             eclipseModel.wtp.component.file.whenMerged { WtpComponent wtpComponent ->
                 if (hasWarOrEarPlugin(project)) {
                     return
                 }
-                wtpComponent.wbModuleEntries.removeAll { wbModule ->
-                    wbModule instanceof WbDependentModule && deleteWtpDependentModule.any { lib -> wbModule.handle.contains(lib) }
-                }
+                adjustWtpComponent(wtpComponent)
             }
         }
 
         project.plugins.withType(WarPlugin) {
             eclipseModel.classpath.containers WEB_LIBS_CONTAINER
 
-            eclipseModel.classpath.file.whenMerged { Classpath classpath ->
-                for (entry in classpath.entries) {
-                    if (entry instanceof AbstractLibrary) {
-                        //this is necessary to avoid annoying warnings upon import to Eclipse
-                        //the .classpath entries can be marked all as non-deployable dependencies
-                        //because the wtp component file declares the deployable dependencies
-                        entry.entryAttributes[AbstractClasspathEntry.COMPONENT_NON_DEPENDENCY_ATTRIBUTE] = ''
-                    }
-                }
+            project.eclipse.classpath.file.whenMerged { Classpath classpath ->
```

Comment:
If my comment above is addressed, these additions would be removed. The container would be the only piece addressed for the WAR plugin.